internal/database/mysql_test.go

```diff
@@ -236,68 +263,98 @@ func TestDeleteUser(t *testing.T) {
 	}
 }
 
-func TestGetUserID(t *testing.T) {
-	subject, probe := setup(t)
-	defer teardown(t, subject, probe)
-
-	t.Run(
-		"Valid_Users", func(t *testing.T) {
-			if err := subject.AddUser("user_id_user", "user_id_user@example.com", "user_id_pass"); err != nil {
-				t.Fatal("Failed to add user -", err)
-			}
-
-			// Get user ID by username
-			id := subject.GetUserID("username", "user_id_user")
-			if id != 1 {
-				t.Error("Failed to get user ID by username")
-			}
-
-			// Get user ID by email
-			id = subject.GetUserID("email", "user_id_user@example.com")
-			if id != 1 {
-				t.Error("Failed to get user ID by email")
-			}
-		},
-	)
-
-	t.Run(
-		"Multiple_Users", func(t *testing.T) {
-			log.Println("Testing for multiple users")
-			if err := subject.AddUser("user2_id_user2", "user2_id_user2@example.com", "user2_id2_pass"); err != nil {
-				t.Fatal("Failed to add user -", err)
-			}
+func TestGetUser(t *testing.T) {
+    subject, probe := setup(t)
+    defer teardown(t, subject, probe)
+
+    // Add a test user to the database:
+    if err := subject.AddUser("test_user", "test_user@example.com", "test_pass"); err != nil {
+        t.Fatal("Failed to add user -", err)
+    }
+
+    testCases := []struct {
+        name       string
+        userID     int
+        expectUser bool
+        username   string
+        email      string
+        password   string
+    }{
+        {"GetExistingUser", 1, true, "test_user", "test_user@example.com", "test_pass"},
+        {"GetNonExistentUser", 999, false, "", "", ""},
+    }
+
+    for _, tc := range testCases {
+        t.Run(tc.name, func(t *testing.T) {
+            user := subject.GetUser(tc.userID)
+            if tc.expectUser {
+                if user == nil {
+                    t.Fatal("Expected to find user, but got nil")
+                }
+                if user.Username != tc.username {
+                    t.Errorf("Expected username to be '%s', but got %s", tc.username, user.Username)
+                }
+                if user.Email != tc.email {
+                    t.Errorf("Expected email to be '%s', but got %s", tc.email, user.Email)
+                }
+                if err := bcrypt.CompareHashAndPassword([]byte(user.Password), []byte(tc.password)); err != nil {
+                    t.Errorf("Password does not match: %v", err)
+                }
+                if user.CreatedOn.IsZero() {
+                    t.Errorf("Expected created_on to be set, but got zero value")
+                }
+                if !user.IsValid() {
+                    t.Errorf("Expected valid user, but got invalid user: %v", user)
+                }
+            } else {
+                if user != nil {
+                    t.Errorf("Expected no user, but got: %v", user)
+                }
+            }
```

Comment:
ditto here, I feel like we could make this code much much shorter and less indented with the other pattern.

Reply:
I refactored the code both ways and it saves 1-2 lines. The refactoring is negligible, and this approach is descriptive and flexible for adding future tests.